src/System.Net.HttpListener/tests/GetContextHelper.cs

```diff
@@ -60,7 +60,15 @@ public class Helpers
     {
         public static void WaitForSocketShutdown(Socket socket)
         {
-            while (SocketConnected(socket));
+            if (PlatformDetection.IsWindows || PlatformDetection.IsOSX)
+            {
+                socket.Shutdown(SocketShutdown.Both);
```

Comment:
@hughbe has commented before on previous commit "These shenanigans are needed because the native implementation is dumb and doesn't know when a socket is closed, but the managed implementation uses sockets directly so this is fine". I think he hypothesizes, that on windows we call into the native apis which does the connections, whereas in managed httplistener we use the socket apis, so the close socket behavior is taking long on windows to terminate connection on other side. But we can unify code and have shutdown and the while check on all platforms, i guess he just did the special case on the platform that was failing consistently.